plonky2x/core/src/frontend/hash/sha/curta/stark.rs

```diff
@@ -0,0 +1,169 @@
+use curta::chip::register::array::ArrayRegister;
+use curta::chip::register::bit::BitRegister;
+use curta::chip::register::element::ElementRegister;
+use curta::chip::register::Register;
+use curta::chip::trace::writer::{InnerWriterData, TraceWriter};
+use curta::chip::Chip;
+use curta::machine::bytes::proof::ByteStarkProof;
+use curta::machine::bytes::stark::ByteStark;
+use curta::math::prelude::*;
+use curta::plonky2::Plonky2Air;
+use itertools::Itertools;
+use log::debug;
+use plonky2::util::timing::TimingTree;
+
+use super::data::{SHAInputData, SHAInputDataValues};
+use super::SHA;
+use crate::frontend::curta::proof::ByteStarkProofVariable;
+use crate::prelude::{CircuitBuilder, PlonkParameters, Variable};
+
+#[derive(Debug, Clone)]
+pub struct SHAStark<
+    L: PlonkParameters<D>,
+    S: SHA<L, D, CYCLE_LEN>,
+    const D: usize,
+    const CYCLE_LEN: usize,
+> {
+    pub stark: ByteStark<S::AirParameters, L::CurtaConfig, D>,
+    pub padded_chunks: Vec<ArrayRegister<S::IntRegister>>,
+    pub end_bits: ArrayRegister<BitRegister>,
```

Comment:
In the future, it would be nice to have precise definitions on what end_bits, digest_bits, and digest_indices are.